### ISSUE TITLE
Sort based on keys

### ISSUE THREAD

**Issue by larsgw** [larsgw/citation.js#43]
Currently, sorting is done by BibTeX label. It would be nice if there's an option to sort as you want, for example first by author, then by year, then by title. Example API could be like this:

```js
const data = Cite(...)

data.sort(['author', 'year', 'title'])
```

**Comment by larsgw** [comment]
Also, sort by custom comparison function.

**Comment by larsgw** [comment]
Now implemented. Will push as soon as DOI is fixed and test cases and docs are ready. Syntax as above:

Regular:
```js
const data = Cite(...)

data.sort(['author', 'year', 'title'])
```

Sort author reversed:
```js
const data = Cite(...)

data.sort(['!author'])
```

Custom callback (same as callback for `Array#sort()`):
```js
const data = Cite(...)

data.sort((objA, objB) => Math.sign(Math.random() - .5))
```

**Comment by larsgw** [comment]
Closed in e528dc919311576658124a7e60e866ecf132e6a8
